### PR TITLE
Anytime main

### PR DESCRIPTION
**Pull request by akmaralAW**
support: 
1. anytime property 
2. extract the output of the optimized solution and add
3. add example case to test it

### CODE REVIEW

File: src/dbcbs_utils.hpp [hunk 1]
```diff
@@ -676,6 +676,45 @@ void export_solutions_joint(const std::vector<LowLevelPlan<dynobench::Trajectory
       *out << joint_actions.at(j).format(dynobench::FMT)<< std::endl;
   }
 }
+
+void extract_motion_primitives(dynobench::Problem problem,
```
Comment (whoenig) [inline src/dbcbs_utils.hpp]: avoid a copy of dynobench::Problem (use const& or &).

File: src/dbcbs_utils.hpp [hunk 2]
```diff
@@ -676,6 +676,45 @@ void export_solutions_joint(const std::vector<LowLevelPlan<dynobench::Trajectory
...
+                              MultiRobotTrajectory &multi_robot_opt_out,
+                              std::map<std::string, std::vector<dynoplan::Motion>> &robot_motions,
+                              const std::vector<std::shared_ptr<dynobench::Model_robot>> &all_robots){
+  
+  size_t robot_idx = 0;
+  for(auto robot_trajectory : multi_robot_opt_out.trajectories) {
+    int idx = 0;
+    int total_actions = robot_trajectory.actions.size();
+    while (idx < total_actions){
+      int num_actions = rand() % 11 + 1; // [0-10]
```
Comment (whoenig) [inline src/dbcbs_utils.hpp]: range should be tunable

File: src/dbcbs_utils.hpp [hunk 3]
```diff
@@ -676,6 +676,45 @@ void export_solutions_joint(const std::vector<LowLevelPlan<dynobench::Trajectory
...
+      num_actions = std::min(num_actions, (total_actions - idx));
+      if ((total_actions - (idx + num_actions)) < 5 && idx + num_actions < total_actions) {
```
Comment (whoenig) [inline src/dbcbs_utils.hpp:694]: what's the magic 5?

File: src/dbcbs_utils.hpp [hunk 4]
```diff
@@ -676,6 +676,45 @@ void export_solutions_joint(const std::vector<LowLevelPlan<dynobench::Trajectory
...
+      std::vector action_vector(robot_trajectory.actions.begin()+idx, robot_trajectory.actions.begin()+idx+num_actions);
+      new_trajectory.actions = action_vector;
+
+      std::vector state_vector(robot_trajectory.states.begin()+idx, robot_trajectory.states.begin()+idx+num_actions+1);
+      // rebase
+      Eigen::VectorXd first_state = state_vector[0];
+      for (auto & state : state_vector) {
+        state[0] -= first_state[0];
+        state[1] -= first_state[1];
+      }
```
Comment (whoenig) [inline src/dbcbs_utils.hpp]: won't work in the 3D case?

**Review by whoenig — APPROVED** [review]

File: src/dbcbs_utils.hpp (hunk 4 above)
Comment (akmaralAW) [reply src/dbcbs_utils.hpp]: fixed

File: src/dbcbs_utils.hpp (hunk 3 above)
Comment (akmaralAW) [reply src/dbcbs_utils.hpp:694]: nothing magic, only ensures that the motions have >=5 length

**Pull request merged**